mobile-widgets/qmlinterface.cpp

```diff
@@ -2,95 +2,98 @@
 #include "qmlinterface.h"
 #include <QQmlEngine>
 
-QMLInterface *QMLInterface::instance()
-{
-	static QMLInterface *self = new QMLInterface;
-	return self;
-}
-
-void QMLInterface::setup(QQmlContext *ct)
+QMLInterface::QMLInterface()
 {
```

Comment:
This is a change I am very uncomfortable with. I had severe problems when testing, which was why I moved the connect to setup. What can indeed be discussed is if setup() should be static.

Also I do not see any special savings by doing it in the constructor.

Reply:
Show us these problems and then we can either
1) fix the root cause or
2) document why these connects can't be done in the constructor as we do it virtually everywhere else in the code base unless there are some very specific reasons.

A valid reason might for example be that the connects are dynamic and can be set from multiple places.

Reply:
I am sorry, I am not here to play, QMLInterface represent quite an investment in time and trial/error from my side and it works. I politely offered my opinion, based on my experience in code written by me, nothing more.

But anyhow this is way over my head, I am just a contributor, while you are a committer, giving us quite different levels in the project. If you feel this is the right thing to do, then please go ahead and merge it.